src/os_posix/os_fallocate.c

```diff
@@ -20,28 +20,27 @@
-        pfh = (WT_FILE_HANDLE_POSIX *)file_handle;                   \
-                                                                     \
-        remap = (offset != pfh->mmap_size);                          \
-        if (remap)                                                   \
-            __wt_prepare_remap_resize_file(file_handle, wt_session); \
-                                                                     \
-        WT_SYSCALL_RETRY(op, ret);                                   \
-        if (remap) {                                                 \
-            if (ret == 0)                                            \
-                __wt_remap_resize_file(file_handle, wt_session);     \
-            else {                                                   \
-                __wt_release_without_remap(file_handle);             \
-                WT_RET(ret);                                         \
-            }                                                        \
-        }                                                            \
-        return (0);                                                  \
+#define WT_CALL_FUNCTION(op)                                                        \
+    do {                                                                            \
+        WT_DECL_RET;                                                                \
+        WT_FILE_HANDLE_POSIX *pfh;                                                  \
+        bool remap;                                                                 \
+                                                                                    \
+        pfh = (WT_FILE_HANDLE_POSIX *)file_handle;                                  \
+                                                                                    \
+        /* Always call prepare. It will return whether a remap is needed or not. */ \
+        __wt_prepare_remap_resize_file(file_handle, wt_session, offset, &remap);    \
+                                                                                    \
+        WT_SYSCALL_RETRY(op, ret);                                                  \
+        if (remap) {                                                                \
+            if (ret == 0)                                                           \
+                __wt_remap_resize_file(file_handle, wt_session);                    \
+            else {                                                                  \
+                __wt_release_without_remap(file_handle);                            \
+                WT_RET(ret);                                                        \
+            }                                                                       \
+        }                                                                           \
+        return (0);                                                                 \
```

Comment:
Does this need to be `return (ret)`. It's not related to your change specifically. But it seems like a potential bug.

If we're not using a memory mapped file and the `fallocate` call fails, it seems we should return the error to the caller. Specifically, `__wt_posix_file_extend` seems to rely on this to determine which `fallocate` method it should use.

Likewise about the similar return from `__posix_file_truncate`, below.

Reply:
Yes, that appears to be a potential bug. I'll look at this and change here too.